st2common/tests/unit/test_util_secrets.py

```diff
@@ -669,3 +701,39 @@ def test_mask_secret_parameters_nested_array_with_object(self):
             ]
         }
         self.assertEqual(expected, result)
+
+    def test_mask_secret_array(self):
```

Comment:
It also wouldn't hurt to add a test case for scenarios where we have an object which is marked "secret: true" at top level, but it also has a sub-schema with different items of which some have "secret: true" attribute. Same goes for array.

I assume in such scenarios, if top level "secret: true" attribute is present, this top level attribute should have a precedence (and we could even ignore / skip iterating over child object / array attributes).